### PR TITLE
Get-DbaServerprotocol use DbaCmObject

### PR DESCRIPTION
**Pull request by PowerDBAKlaas**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [x] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Pester test is included

### Purpose
 
- Reuse functionality of Get-DbaCmObject
- use messaging standard
- improve speed
### Approach

- replace `Get-CimInstance` with `Get-DbaCmObject`
- replace `Write-Verbose` with `Write-Message`

### CODE REVIEW

**Comment by potatoqualitee** [comment]
@PowerDBAKlaas - I wrote a Test for this command that passes 👍 But I don't like the output from the new Cm commands and would like to get that addressed prior to merging.

@FriedrichWeinmann - I submitted a PR to remove the exclamation point but will probably need to up the library's version #1884. Either way, the command works, but it complains, seemingly unnecessarily

image

**Comment by potatoqualitee** [comment]
Will wait to merge with the other one

**Comment by wsmelton** [comment]
If you don't like the output of `Get-DbaCmObject` then it should be addressed in that command. But what output are you talking about?

**Comment by potatoqualitee** [comment]
Could not find valid protocol warning at the top that comes from the cm command

**Comment by wsmelton** [comment]
If you issue that same command with Verbose and post that output?

**Comment by wsmelton** [comment]
The stop function in that line show set to continue so yeah the phrase in that message is inaccurate.

**Comment by FriedrichWeinmann** [comment]
Well now ... that's mightily odd. Looks like I seriously need some quality time with `Get-DbaCmObject`, both for the performance issue, as well as this. Obviously, throwing an exception _and_ returning the requested information is a bit contradictory.
Looking into it.

File: functions/Get-DbaServerProtocol.ps1 [hunk 1]
```diff
@@ -55,12 +58,12 @@ Param (
   [parameter(ValueFromPipeline)]
   [Alias("cn","host","Server")]
   [string[]]$ComputerName = $env:COMPUTERNAME,
```
Comment (wsmelton) [inline functions/Get-DbaServerProtocol.ps1]: Need to set this to `DbaInstanceParameter`

File: functions/Get-DbaServerProtocol.ps1 [hunk 2]
```diff
@@ -71,67 +74,33 @@ PROCESS
           if ( $Server.ComputerName )
           {
               $Computer = $server.ComputerName
```
Comment (wsmelton) [inline functions/Get-DbaServerProtocol.ps1]: Same question as other PR do we need to add a `Resolve-DbaNetworkName`?

Comment (potatoqualitee) [inline functions/Get-DbaServerProtocol.ps1]: looks like someone added it 💃

**Comment by potatoqualitee** [comment]
merging to prevent future conflicts. a release or two with slower commands is less headache 🙉

**Pull request merged**